### ISSUE TITLE
CLI bin console style enhancement.

### ISSUE THREAD

**Issue by ulivz** [marko-js/marko#793]
## Summary
The many popular framework's CLI(`command line interface`) is pretty accessible with a good-formatted and colorful-suitable, for example:

![image](https://user-images.githubusercontent.com/23133919/29083004-dc2b2bb4-7c99-11e7-832d-89a3685a75a4.png)

As a comparison, which is our snapshot:

![image](https://user-images.githubusercontent.com/23133919/29083071-11c5e7fa-7c9a-11e7-88da-cac381606036.png)
 
### Context
No very important for me, but I think it can be a power which cannot be ignored to attract more users.

### Possible Implementation
I have listed some awesome libraries to quickly build awesome CLI:

[`cac`](https://github.com/egoist/cac)
[`commander.js`](https://github.com/tj/commander.js)
[`yargs`](https://github.com/yargs/yargs)
[`caporal.js`](https://github.com/mattallty/Caporal.js)
[`meow`](https://github.com/sindresorhus/meow)

### Is this something you're interested in working on?
Yes, I'm very interested in doing this due to I have woking for CLi creation for some time, it's so attracted for me.

**Comment by patrick-steele-idem** [comment]
I agree that we can improve the output of the CLI. We are using [argly](https://github.com/patrick-steele-idem/argly) because in the past I did not like the API that I found in similar libraries. It's been awhile so I can't remember if I have checked all of the other ones you listed. I think it would be fairly straightforward to enhance `argly` with improved formatting and colorization and that would be my preference (vs switching to a different CLI library).

The code for outputting the usage string can be found [here](https://github.com/patrick-steele-idem/argly/blob/13b186ee10e710e19a5c600d89946958c5d73db9/src/Parser.js#L252-L311).

**Comment by ulivz** [comment]
@patrick-steele-idem I think I can work for this enhancement with straightforward to enhance `argly`.
